framework/src/play/jobs/Job.java

```diff
@@ -164,7 +164,19 @@ public void every(String delay) {
      *            time in seconds
      */
     public void every(int seconds) {
-        JobsPlugin.executor.scheduleWithFixedDelay(this, seconds, seconds, TimeUnit.SECONDS);
+        // Wrap this job in a Runnable that catches and ignores all thrown exception.
+        // If there's an unhandled exception, then the executor would suppress all
+        // further executions, violating the contract.  Note that this does
+        // not interfere with the normal handling of onException().
+        Runnable neverThrowingRunnable = new Runnable() {
```

Comment:
> fix just the commit 45d390d on line 138 which introduced the error (by using a flag, check now and in too)

Thanks for the extra clarification.  I'll rework my pull request to do this, following the pattern of your Gist, but applied to `now()` and `in()` instead of `every()`.  It may take me a while to put the update together.